### PR TITLE
adds quic receiver to shred-fetch-stage

### PR DESCRIPTION
**Pull request by behzadnouri**
#### Problem
Working towards migrating turbine to QUIC.

#### Summary of Changes
Added QUIC receiver to shred-fetch-stage.

### CODE REVIEW

**Comment by t-nelson** [comment]
is this something that can wait for 1.16 to be cut?  really need to keep branch diffs to a minimum for the time being

**Comment by behzadnouri** [comment]
> is this something that can wait for 1.16 to be cut? really need to keep branch diffs to a minimum for the time being

I actually need this to be deployed to testnet asap,
so that we can test performance on an actual cluster and optimize where needed.

**Comment by behzadnouri** [comment]
Can I have this reviewed/merged?
I also have https://github.com/solana-labs/solana/pull/31610 depending on this chain.

Also these PRs only add a code-path which is inactive for all shreds. The goal is to incrementally rollout and monitor for small percent of shreds on testnet and debug/optimize based on the outcome.

**Comment by t-nelson** [comment]
we teed this feature set up for discussion on the thursday priorities call.  general consensus today was that holding quic turbine for 1.16 would fair better for 1.16 stabilization

File: core/src/shred_fetch_stage.rs [hunk 1]
```diff
@@ -23,6 +26,12 @@ use {
     },
 };
 
+const MAX_QUIC_CONNECTIONS_PER_PEER: usize = 8;
+const MAX_STAKED_QUIC_CONNECTIONS: usize = 2000;
+const MAX_UNSTAKED_QUIC_CONNECTIONS: usize = 1000;
+const QUIC_WAIT_FOR_CHUNK_TIMEOUT: Duration = Duration::from_secs(5);
+const QUIC_COALESCE_WAIT: Duration = Duration::from_millis(10);
```
[the remark this answers is not in the document]
Comment (behzadnouri) [reply core/src/shred_fetch_stage.rs:33]: > Is the idea to define values here so we can tune TVU independently from TPU 

yes

**Comment by behzadnouri** [comment]
> In order to try to meet in the middle, maybe something like [a5f290a](https://github.com/solana-labs/solana/commit/a5f290a66f10a25e0c3ee004a1fcd2889eca2785) would be agreeable? That is, push the code in but not explicitly spawn the server by default for now.

We actually need to start testing QUIC on testnet for propagating a small percentage of slots.

**Review by steviez — APPROVED** [review]
With v1.16 branch cut, I believe we're good to go on these.

I see there have been some conversations in Discord and a SIMD, but assuming you're looking to push this in and test in parallel while those get hashed out ?

**Pull request merged**